app/vendor/views.py

```diff
@@ -48,6 +50,53 @@ def new_listing():
     return render_template('vendor/new_listing.html', form=form)
 
 
+@vendor.route('/csv-upload', methods=['GET', 'POST'])
+@login_required
+@vendor_required
+def csv_upload():
+    """Create a new item."""
+    form = NewCSVForm()
+    listings = []
+    if form.validate_on_submit():
+        csv_field = form.file_upload
+        buff = csv_field.data.stream
+        csv_data = csv.DictReader(buff, delimiter=',')
+        #for each row in csv, create a listing
+        current_vendor = Vendor.get_vendor_by_user_id(user_id=current_user.id)
+        for row in csv_data:
+            #cheap way to skip weird 'categorical' lines
+            if (row[current_vendor.product_id_col]).strip().isdigit():
+                safe_price = stripPriceHelper(row[current_vendor.price_col])
+                proposed_listing = Listing.add_csv_row_as_listing(csv_row=row, price=safe_price)
+                queried_listing = Listing.get_listing_by_product_id(product_id=row[current_vendor.product_id_col])
+                if queried_listing:
+                    # case: listing exists and price has not changed
+                    if queried_listing.price == float(safe_price):
+                        listings.append(proposed_listing)
```

Comment:
Do we really need to render all the items that aren't changed? i think we should probably just show deleted and changed items